### PR TITLE
perf: cleanup codec logic

### PR DESCRIPTION
**Pull request by tsi**
* Cleanup codec logic
* Update to public `videojs-contrib-dash`, since it is maintained again.

### CODE REVIEW

File: src/plugins/cloudinary/index.js [hunk 1]
```diff
@@ -86,8 +83,8 @@ class CloudinaryContext {
       _source = src;
 
       const isDashRequired = options.sourceTypes && options.sourceTypes.some(s => s.includes('dash'));
-      if (plugins.dashPlugin && isDashRequired) {
-        plugins.dashPlugin().then(() => {
+      if (isDashRequired) {
```
Comment (ehab-cl) [inline src/plugins/cloudinary/index.js:86]: 🕺

File: src/plugins/cloudinary/models/video-source/video-source.utils.js [hunk 2]
```diff
@@ -1,36 +1,38 @@
...
   if (!res) {
-    res = format.split('\/').shift();
+    res = format.split('/').shift();
   }
 
   return res;
 }
 
-const isContainCodec = (value) => value && some(Object.values(VIDEO_CODEC), (codec) => value.includes(codec));
+const isContainCodec = value =>
```
Comment (ehab-cl) [inline src/plugins/cloudinary/models/video-source/video-source.utils.js:31]:
```suggestion
const doesContainCodec = value =>
```

Comment (ehab-cl) [reply src/plugins/cloudinary/models/video-source/video-source.utils.js:31]: or just `hasCodec` maybe

File: src/plugins/cloudinary/models/video-source/video-source.utils.js [hunk 3]
```diff
@@ -1,36 +1,38 @@
...
+  value && some(Object.values(VIDEO_CODEC), codec => value.includes(codec));
 
-const hasCodecSrcTrans = (transformations) => some(['video_codec', 'streaming_profile'], (key) => isKeyInTransformation(transformations, key));
+const hasCodecSrcTrans = transformations =>
+  some(['video_codec', 'streaming_profile'], key => isKeyInTransformation(transformations, key));
```
Comment (ehab-cl) [inline src/plugins/cloudinary/models/video-source/video-source.utils.js:35]: For later or maybe even now you can just refactor this usage of some and use the es6 built in some method for arrays.

**Review by ehab-cl — APPROVED** [review]
💪

File: src/plugins/cloudinary/common.js [hunk 4]
```diff
@@ -102,109 +102,13 @@ const cloudinaryErrorsConverter = ({ errorMsg, publicId, cloudName, statusCode }
   return error;
 };
 
-const codecShorthandTrans = short => {
```
[the remark this answers is not in the document]
Comment (tsi) [reply src/plugins/cloudinary/common.js:105]: We decided not to test for specific codec support anymore, since almost all browsers now support them all.
With the default `f_auto` there shouldn't be an issue ever.
There are some edge cases in which old Safari's for example might try playing an AV1 and will have to fall-back to the next `sourceType`, which will cause a console error but nothing fatal. And if that's a concern we leace it to the implementor to prevent.
